addon/components/sl-menu.js

```diff
@@ -338,13 +341,28 @@ export default Ember.Component.extend({
     children: null,
 
     /**
-     * Additional class names to be added to the classNames attribute
+     * Array of classes to be added to the element's class attribute
      *
      * @type {String[]}
      */
     extraClassNames: [],
 
     /**
+     * Additional class string to be added to the element's class attribute
+     *
+     * @type {String}
+     */
+    extraClassNamesString: function() {
+        var extraClassNames = this.get( 'extraClassNames' );
+
+        if ( !Ember.isNone( extraClassNames ) ) {
+            return extraClassNames.join( ' ' );
+        } else {
+            return [];
```

Comment:
Cool.

The preferred pattern at that point then is to define a return value at the top and return it at the end of the function.  The if statement can modify it as appropriate.  Falls into the "only one return statement per method" recommendation.